src/lib.rs

```diff
@@ -262,19 +291,62 @@ impl Scope {
             .or_insert_with(|| Import::new(path, ty))
     }
 
+    /// Returns true if a module named `name` exists in this scope.
+    pub fn contains_module<Q: ?Sized>(&self, name: &Q) -> bool 
+    where
+        Q: Hash + ordermap::Equivalent<RcKey>
+    {
+        self.modules.contains_key(name)
+    }
+
     /// Push a new module definition, returning a mutable reference to it.
+    /// 
+    /// # Note
+    /// 
+    /// Since a module's name must uniquely identify it within the scope in 
+    /// which it is defined, calling this function with a name already used 
+    /// in this scope will (silently) clobber the existing module definition.
+    /// 
+    /// In many cases, the [`module_or_add`] function is preferrable, as it will
+    /// return the existing definition instead.
+    /// 
+    /// [`module_or_add`]: #method.module_or_add
     pub fn new_module(&mut self, name: &str) -> &mut Module {
         self.push_module(Module::new(name));
 
-        match *self.items.last_mut().unwrap() {
-            Item::Module(ref mut v) => v,
-            _ => unreachable!(),
+        &mut self.modules[name]
+    }
+
+    /// Returns a mutable reference to a module if it is exists in this scope. 
+    pub fn module_mut(&mut self, name: &str) -> Option<&mut Module> {
+        self.modules.get_mut(name)
+    }
+
+    /// Returns a mutable reference to a module, creating it if it does 
+    /// not exist.
+    pub fn module_or_add(&mut self, name: &str) -> &mut Module {
+        if self.modules.contains_key(name) {
+            &mut self.modules[name]
+        } else {
+            self.new_module(name)
         }
     }
 
     /// Push a module definition.
+    /// 
+    /// # Note
+    /// 
+    /// Since a module's name must uniquely identify it within the scope in 
+    /// which it is defined, pushing a module whose name is already defined
+    /// in this scope will clobber the existing module definition.
+    /// 
+    /// In many cases, the [`module_or_add`] function is preferrable, as it will
+    /// return the existing definition instead.
+    /// 
+    /// [`module_or_add`]: #method.module_or_add
     pub fn push_module(&mut self, item: Module) -> &mut Self {
```

Comment:
This probably should panic if called and there already is a module with the same name in the scope.